File: Android/scenes/ktv_singbattle/src/main/java/io/agora/scene/ktv/singbattle/widget/UserLeaveSeatMenuDialog.java
package io.agora.scene.ktv.singbattle.widget;

import android.content.Context;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;

import com.bumptech.glide.Glide;

import io.agora.scene.base.component.BaseDialog;
import io.agora.scene.base.utils.KtExtendKt;
import io.agora.scene.ktv.singbattle.databinding.KtvSingbattleDialogUserSeatMenuBinding;

/**
 * Room user menu             
 */
public class UserLeaveSeatMenuDialog extends BaseDialog<KtvSingbattleDialogUserSeatMenuBinding> {
    public UserLeaveSeatMenuDialog(@NonNull Context context) {
        super(context);
    }

    @NonNull
    @Override
    protected KtvSingbattleDialogUserSeatMenuBinding getViewBinding(@NonNull LayoutInflater layoutInflater) {
        return KtvSingbattleDialogUserSeatMenuBinding.inflate(layoutInflater);
    }

    public void setAgoraMember(String name, String headUrl) {
        getBinding().tvName.setText(name);
        Glide.with(getContext())
                .load(headUrl).error(io.agora.scene.widget.R.mipmap.default_user_avatar)
                .into(getBinding().ivUser);
    }

    @Override
    public void initView() {
        setCanceledOnTouchOutside(true);
        getWindow().setWindowAnimations(io.agora.scene.widget.R.style.popup_window_style_bottom);
        getBinding().btSeatoff.setOnClickListener(this::seatOff);
        getBinding().btLeaveChorus.setOnClickListener(this::leaveChorus);
    }

    private void seatOff(View v) {
        if (getOnButtonClickListener() != null) {
            dismiss();
            getOnButtonClickListener().onRightButtonClick();
        }
    }

    private void leaveChorus(View v) {
        if (getOnButtonClickListener() != null) {
            dismiss();
            getOnButtonClickListener().onLeftButtonClick();
        }
    }

    @Override
    protected void setGravity() {
        getWindow().setLayout(
                ViewGroup.LayoutParams.MATCH_PARENT,
                (int) KtExtendKt.getDp(220)
        );
        getWindow().getAttributes().gravity = Gravity.BOTTOM;
    }
}

File: Android/scenes/cantata/src/main/java/io/agora/scene/cantata/ui/dialog/UserLeaveSeatMenuDialog.kt
package io.agora.scene.cantata.ui.dialog

import android.content.Context
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import com.bumptech.glide.Glide
import io.agora.scene.base.component.BaseDialog
import io.agora.scene.cantata.R
import io.agora.scene.cantata.databinding.CantataDialogUserSeatMenuBinding
import io.agora.scene.cantata.ui.widget.OnClickJackingListener

/**
 * 房间用户菜单
 */
class UserLeaveSeatMenuDialog constructor(context: Context) : BaseDialog<CantataDialogUserSeatMenuBinding?>(context) {
    override fun getViewBinding(layoutInflater: LayoutInflater): CantataDialogUserSeatMenuBinding {
        return CantataDialogUserSeatMenuBinding.inflate(layoutInflater)
    }

    fun setAgoraMember(name: String?, headUrl: String?) {
        binding!!.tvName.text = name
        Glide.with(context)
            .load(headUrl).error(R.mipmap.userimage)
            .into(binding!!.ivUser)
    }

    public override fun initView() {
        setCanceledOnTouchOutside(true)
        window?.setWindowAnimations(R.style.popup_window_style_bottom)
        binding?.apply {
            btSeatoff.setOnClickListener(object :OnClickJackingListener(){
                override fun onClickJacking(view: View) {
                    seatOff(view)
                }
            })
            btLeaveChorus.setOnClickListener(object :OnClickJackingListener(){
                override fun onClickJacking(view: View) {
                    leaveChorus(view)
                }
            })
        }
    }

    private fun seatOff(v: View) {
        if (getOnButtonClickListener() != null) {
            dismiss()
            getOnButtonClickListener().onRightButtonClick()
        }
    }

    private fun leaveChorus(v: View) {
        if (getOnButtonClickListener() != null) {
            dismiss()
            getOnButtonClickListener().onLeftButtonClick()
        }
    }

    override fun setGravity() {
        window!!.attributes.gravity = Gravity.BOTTOM
    }
}